### PR TITLE
Issue 1466/event date pickers

### LINKED ISSUE #1466: Time changes when event is edited
## Description
A user (Adam) reports the following bug, but was unable to reproduce it when 

## Steps to reproduce
1. Go to http://localhost:3000/organize/1/projects/1
2. Click on Create Activity
3. Click on "Event"
4. In the new event, add a start and end time
5. Edit the name of the event at the top of the screen (it might be that this must happen directly after step 4, without clicking outside of the form in 4)

### Expected Behaviour
Save event times and text.

### Actual Behaviour
Event times are reset when text is saved.

## Screenshots (if you have any)
None

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR updates the date- and time picker components used in the `EventOverviewCard`, and refactors to simplify styling. It also updates styles and types in the smart search `TimeFrame` component

## Screenshots
the only visual changes are of  the TimeFrame: 
was 
bild
now
bild

## Changes
* Changes to MUI 6 `TimeField` components 
* Refactors state handling to use strings instead of dayjs objects. 
* Adds limitiations to the end date picker so you can only pick end dates after the start date
* Adds limitations to the end time picker so you can only pick end times that are at least 1 minute after the start time.
* Updates the styling of the `TimeFrame` component to match the smart search scentence look.
* Updates to use `dayjs` format in the `DatePicker` in the `TimeFrame` component

## Notes to reviewer
Don't forget to `yarn install` ;)

## Related issues
Resolves #1466

### CODE REVIEW

**Comment by richardolsson** [comment]
To whomever reviews this, please double check that localization is working correctly for these new pickers, i.e. that month names, date formats etc are displayed in the local language (the same as other parts of the UI), e.g. Swedish, Norwegian or German.

**Review by kaulfield23 — COMMENTED** [review]
Nice work! I really love how it makes code shorter than before! Especially with `DatePicker` component! I checked if it works well in other languages and everything works fine! Codes, I spent more than half of the day to understand everything. I think it looks good in my eyes but I need @richardolsson 's opinion as well!

File: src/features/smartSearch/components/inputs/StyledDatePicker.tsx [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+        textField: {
+          inputProps: {
+            sx: {
+              fontSize: theme.typography.h4.fontSize,
+              padding: 0,
+            },
+          },
+          sx: {
+            paddingRight: 1,
+            paddingTop: '1px',
```
[the remark this answers is not in the document]
Comment (ziggabyte) [reply src/features/smartSearch/components/inputs/StyledDatePicker.tsx:22]: Yeah, it's not pretty but it's the solution i found that made it look right, so I guess it works for now?

**Review by richardolsson — APPROVED** [review]
I had a look and also tried running the code and I think it looks good! 💯

**Comment by richardolsson** [comment]
Oops, looks like there is a bit of a nasty conflict. Will you have a look at it @ziggabyte ?

**Review by richardolsson — APPROVED** [review]
Still looks good after resolving conflicts. Ready to merge!

**Pull request merged**